### DISCUSSION TITLE
How to auto-add date to new pages?

### DISCUSSION THREAD

**Discussion by poblabs** [estruyf/vscode-front-matter#252] [Q&A] [answered]
Hey everyone. I'm trying to see if there's a way to auto-add the creation date to new pages without having to click the "Now" button in the sidebar? 

The lastmod is working great, but I don't see a way to have creation date automatically added to the front matter. 

Thanks!

**Comment by zivbk1** [comment, marked as answer]
Please see... https://frontmatter.codes/docs/content-types#placeholders

**Reply by poblabs** [reply]
I'm still a bit lost after reading that whole page. When I add the examples to frontmatter.json, or make my own frontmatter and "create new template from file"- when I do new file, name it helloworld.md, there is no front matter generated automatically.

**Reply by estruyf** [reply]
@poblabs would you be able to share the contents of your `frontmatter.json` file?

**Reply by poblabs** [reply]
@estruyf Sure, it's very generic since I'm just starting to understand frontmatter 😄 

```
{
  "$schema": "https://frontmatter.codes/frontmatter.schema.json",
  "frontMatter.framework.id": "hugo",
  "frontMatter.content.publicFolder": "static",
  "frontMatter.content.placeholders": [
    {
      "id": "permalink",
      "value": "/blog/{{slug}}.html"
    }
  ]
}
```

I think what I'm expecting is when I click the new file button, name it `helloworld.md`, the front matter to get pre-populated automatically, then I can work on the page. 

![image](https://user-images.githubusercontent.com/3484775/154316934-5011c553-a6a5-4a76-9ad4-636235c6d841.png)

Something like: 

```
---
title: hello world
slug: hello-world
description: null
author: null
date: '2022-02-16 11:57:54 -05:00'
lastmod: null
draft: true
tags: []
categories: []
---
```

But currently, when I click that new file button, there is no front matter being generated. Same results when trying these options: 

![image](https://user-images.githubusercontent.com/3484775/154317639-405ac4a6-84ca-4328-8a5e-d35e8b552ca8.png)

I also found [these actions to start a new article](https://frontmatter.codes/docs/commands#overview), but it doesn't match what I'm seeing:

![image](https://user-images.githubusercontent.com/3484775/154317874-78230331-9df2-406c-8a73-e7aceebb0189.png)

I'm 100% sure the problem is with me, just a bit lost right now on the proper workflow.

**Reply by zivbk1** [reply]
You will want to create a template, in that template you can put placeholders. 
https://frontmatter.codes/docs/content-types#creating-a-template

Then in the 'Dashboard' you can create new content from that template and it should put it in the right spot and fill in the placeholders with the information that you want.

**Reply by poblabs** [reply]
I've created a template, I've registered the folders, but still don't see a way to create new content from the dashboard? 
![image](https://user-images.githubusercontent.com/3484775/154321363-f20f24f5-d718-4a5c-ba12-b9244c60d14d.png)

**Reply by zivbk1** [reply]
![image](https://user-images.githubusercontent.com/6154767/154321507-e3a3d8f4-4381-4ba8-9b0b-140722fe9415.png)

@estruyf it seems that anchor links are not working consistently for me so I am adding this screenshot of the relevant document.

**Reply by zivbk1** [reply]
@poblabs that folder structure doesn't look quite right to me.  Do you have time for a zoom chat to go through it?

**Reply by poblabs** [reply]
Yeah I've been messing around - probably not the best screenshot. Also, yep available for next 30m

**Reply by zivbk1** [reply]
> Yeah I've been messing around - probably not the best screenshot. Also, yep available for next 30m

Please email klein@thunderheadeng.com and I'll send you a zoom link.

**Reply by zivbk1** [reply]
Sorry, just updated my email, had a typo... it is correct now.

**Reply by estruyf** [reply]
thanks, @zivbk1 for jumping in ❤️.

**Reply by estruyf** [reply]
btw @poblabs the `Front Matter: New Article` was an old command. Will be my fault that you used it, as I didn't update the image in the docs. Just did this on the beta docs.

**Reply by poblabs** [reply]
@zivbk1 Thanks for the help! I'm now capable enough with Front Matter to get myself into more trouble 😄

**Reply by zivbk1** [reply]
@estruyf Pat and I had a quick chat and got it sorted out.

**Reply by estruyf** [reply]
@zivbk1 also found the linking issue, it had to do with the code blocks and their syntax highlighting. Pushed out a fix.
